adminSiteClient/MultiDimIndexPage.tsx

```diff
@@ -0,0 +1,282 @@
+        },
+    ]
+}
+
+function deserializeMultiDim(mdim: ApiMultiDim): MultiDim {
+    return {
+        ...mdim,
+        updatedAt: new Date(mdim.updatedAt),
+    }
+}
+
+async function fetchMultiDims(admin: Admin) {
+    const { multiDims } = await admin.getJSON<{ multiDims: ApiMultiDim[] }>(
+        "/api/multi-dims.json"
+    )
+    return multiDims.map(deserializeMultiDim)
+}
+
+async function patchMultiDim(admin: Admin, id: number, data: Json) {
+    const { multiDim } = await admin.requestJSON<{ multiDim: ApiMultiDim }>(
+        `/api/multi-dims/${id}`,
+        data,
+        "PATCH"
+    )
+    return deserializeMultiDim(multiDim)
+}
+
+const NotificationContext = createContext(null)
+
+export function MultiDimIndexPage() {
+    const { admin } = useContext(AdminAppContext)
+    const [notificationApi, notificationContextHolder] =
+        notification.useNotification()
+    const [search, setSearch] = useState("")
+    const queryClient = useQueryClient()
+
+    useEffect(() => {
+        admin.loadingIndicatorSetting = "off"
+        return () => {
+            admin.loadingIndicatorSetting = "default"
+        }
+    }, [admin])
+
+    const { data } = useQuery({
+        queryKey: ["multiDims"],
+        queryFn: () => fetchMultiDims(admin),
+    })
+
+    const publishMutation = useMutation({
+        mutationFn: ({ id, published }: { id: number; published: boolean }) =>
+            patchMultiDim(admin, id, { published }),
+        onSuccess: async ({ published }) => {
+            notificationApi.info({
+                message: published ? "Published" : "Unpublished",
+                description: <Link to="/deploys">Check deploy progress</Link>,
+                placement: "bottomRight",
+            })
+            return queryClient.invalidateQueries({ queryKey: ["multiDims"] })
+        },
+    })
+
+    const slugMutation = useMutation({
+        mutationFn: ({ id, slug }: { id: number; slug: string }) =>
+            patchMultiDim(admin, id, { slug }),
+        onSuccess: async ({ published }) => {
+            notificationApi.info({
+                message: "Slug updated",
+                description: published ? (
+                    <Link to="/deploys">Check deploy progress</Link>
+                ) : undefined,
+                placement: "bottomRight",
+            })
+            return queryClient.invalidateQueries({ queryKey: ["multiDims"] })
+        },
+    })
+
+    const filteredMdims = useMemo(() => {
+        const query = search.trim().toLowerCase()
+        return data?.filter((mdim) =>
+            [mdim.title, mdim.slug].some((field) =>
+                field.toLowerCase().includes(query)
+            )
+        )
+    }, [data, search])
+
+    const columns = createColumns(slugMutation, publishMutation)
+
+    return (
+        <AdminLayout title="Multidimensional Data Pages">
+            <NotificationContext.Provider value={null}>
```

Comment:
I don't think we need this extra context, since it's always null and not antd-specific anyhow?